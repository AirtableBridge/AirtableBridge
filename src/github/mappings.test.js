const { mapIssue, mapCommit, mapUser, mapPR } = require("./mappings");

function fixture(name) {
  return require(`../fixtures/${name}.json`);
}

function snap(data) {
  // console.log(data);
  expect(data).toMatchSnapshot();
}

describe("mappings", () => {
  it("issue", () => snap(mapIssue(fixture("issue"))));
  it("commit", () => snap(mapCommit(fixture("commit"))));
  it("user", () => snap(mapUser(fixture("user"))));
  it("pr", () => snap(mapPR(fixture("pr"))));
});
