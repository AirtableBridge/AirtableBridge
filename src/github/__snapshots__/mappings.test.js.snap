// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mappings commit 1`] = `
Object {
  "Author": "tohmjudson",
  "Committer": "jasonLaster",
  "Date": "10/21/2017",
  "Message": "Blackboxed sources distinguished by icon in source tree  (#4441)",
  "Sha": "1ded1939f224265a585c020f4b7c608b14dc0af7",
  "Url": "https://github.com/devtools-html/debugger.html/commit/1ded1939f224265a585c020f4b7c608b14dc0af7",
}
`;

exports[`mappings issue 1`] = `
Object {
  "Assignees": Array [
    "belld19233",
  ],
  "Author": "belld19233",
  "Closed": null,
  "Created": "10/16/2017",
  "Description": "Ignore this bug. Instead of making comments on bugs in progress, I will do so here in order to debug the issue of Probot not detecting some /claim comments.

Also, this will test our Probot's ability to assign users to a bug.",
  "ID": 4384,
  "Labels": Array [
    "in progress",
  ],
  "Number": 4384,
  "Open": true,
  "Org": "devtools-html",
  "Repository": "debugger.html",
  "Title": "Debugging Probot",
  "URL": "https://github.com/devtools-html/debugger.html/issues/4384",
  "Updated": "10/20/2017",
}
`;

exports[`mappings pr 1`] = `
Object {
  "Author": "GarethSharpe",
  "Avatar": Array [
    Object {
      "url": "https://avatars1.githubusercontent.com/u/22062107?v=4",
    },
  ],
  "Branch": "GarethSharpe:master",
  "Closed": null,
  "Created": "10/23/2017",
  "Description": "Associated Issue: #4410 

### Summary of Changes

* Created breakpoint dropdown styling 
* Breakpoint toggle returns dropdown
* Changed dropdown .svg to \\"+\\"

Example test plan:

- [x] Create an arbitrary breakpoint
- [x] Breakpoint window displays panel toggle 
- [x] Clicking “+” opens the panel

TODO:
* Adjust dropdown width to fit parent
* Add menu items (with corresponding images)
* Add menu item implementation
* Remove CommandBar options as needed",
  "ID": 4478,
  "Merged": null,
  "Open": true,
  "Title": "Setup breakpoint dropdown",
  "Updated": "10/23/2017",
  "Url": "https://github.com/devtools-html/debugger.html/pull/4478",
}
`;

exports[`mappings user 1`] = `
Object {
  "Avatar": Array [
    Object {
      "url": "https://avatars0.githubusercontent.com/u/25886486?v=4",
    },
  ],
  "Company": null,
  "Email": null,
  "Location": null,
  "Login": "belld19233",
  "Name": "Dave B",
  "Profile": "https://github.com/belld19233",
}
`;
